packages/bundle-source/cache.js

```diff
@@ -65,12 +79,20 @@ export const makeFileReader = (fileName, { fs, path }) => {
  *     },
  *   path: Pick<import('path'), 'resolve' | 'relative' | 'normalize'>,
  * }} io
+ * @param {number} pid
  */
-export const makeFileWriter = (fileName, { fs, path }) => {
+export const makeFileWriter = (fileName, { fs, path }, pid) => {
   const make = there => makeFileWriter(there, { fs, path });
   return harden({
     toString: () => fileName,
     writeText: (txt, opts) => fs.promises.writeFile(fileName, txt, opts),
+    atomicWriteText: async (txt, opts) => {
```

Comment:
do we need the `pid` for the life of the file writer? or could it be an arg to `atomicWriteText`?

Reply:
I considered making thee `pid` a parameter of `atomicWriteText` but, given that it is guaranteed to be consistent, it’s the only value guaranteed to work for the lifetime of the process, even if there’s trash on the file system from a previous boot with a coïncident pid, and that threading `pid` to all the call sites is a little invasive, I went with capturing it early.

Reply:
Maybe for later when fs graduates to a pkg... It seems burdensome to provide a pid to make a file writer. Maybe a maker without pid and without atomicWrite.

Reply:
Agreed. Can be another layer.

Reply:
Posted a commit that makes `makeAtomicFileWriter` a separable layer that extends `makeFileWriter`.

I committed a minor crime in applying a pattern for maker extensibility (the same crime committed in `makeCompartmentConstructor`).